### PR TITLE
docs: update PUBLISHING instructions with git flow

### PR DESCRIPTION
**Pull request by anniehu4**
### Summary:

Add updated PUBLISH instructions with our new main -> next -> release branch model 

### Test Plan:
I tested this out with a couple of test branches, so I'm hopeful this strategy avoids nastiness happening to `next` that forces everyone to re-create PRs and stuff

### CODE REVIEW

File: docs/PUBLISHING.md [hunk 1]
```diff
@@ -17,18 +17,24 @@ Once we publish major version 1, we will begin following conventional semver.
...
+- release branches are merged back into `main` and `next` when ready
+
+#### Cutting a release branch
 
 1. Confirm that all checks are green on CI.
-2. Run `git checkout main`.
-3. Run `yarn` to ensure all dependencies are upto date.
+2. Run `git checkout next && git pull origin next` to ensure you have the most up-to-date changes.
+3. Run `git checkout -b release-v<version>`
 4. Run
```
Comment (dierat) [inline docs/PUBLISHING.md:38]:
```suggestion
4. Run:
```
Every time I read this I think the 4th step is to literally run away 🏃

File: docs/PUBLISHING.md [hunk 2]
```diff
@@ -17,18 +17,24 @@ Once we publish major version 1, we will begin following conventional semver.
...
 _Before the first time you publish_, make sure to:
 
 - set up Two Factor Authentication for your npm account
 - run `npm login` in your terminal to generate an access token for publishing
 
-#### Publishing steps
+### Publishing steps
+We follow a [git-flow](https://nvie.com/posts/a-successful-git-branching-model/) based model, where:
+- development happens on `next`
+- release branches are cut from `next`. only bugfixes and documentation updates can be added to release branches.
```
Comment (ahuth) [inline docs/PUBLISHING.md:30]:
```suggestion
- release branches are cut from `next`. Only bugfixes and documentation updates can be added to release branches.
```

[the remark this answers is not in the document]
Comment (ahuth) [reply docs/PUBLISHING.md:30]: nvm, the rest of the doc goes into this.

File: docs/PUBLISHING.md [hunk 3]
```diff
@@ -40,13 +46,23 @@ yarn release:breaking
...
-git push --follow-tags origin <branch> && npm publish
+git push --follow-tags origin <branch>
 ```
 
-6. Communicate the changes via all appropriate channels (if this is a breaking package update that the broader team needs to know about):
+#### Merging a release branch
+6. Open a pull request for the release branch, merging into **`main`**. Merge the PR through a **merge commit**:
+![Screen Shot 2022-05-26 at 10 56 20 AM](https://user-images.githubusercontent.com/15840841/170514789-4f936ba2-c63d-486c-827a-b9e9e86b612e.png)
+
+This is important to ensure `next` and `main` stay in sync!
```
[the remark this answers is not in the document]
Comment (anniehu4) [reply docs/PUBLISHING.md:57]: The main intention is to keep clear releases while allowing for long-running work on the `next` branch, like for a pilot. It was a suggestion from BigMedium, and we agreed to try it out for a bit 🤷‍♀️  So far there have been some improvements:
- `main` (and thus storybook) always matches the state of the latest published package
- It's clearer what the next release will be / there's a clear branch for testing it altogether

but also some downsides, mainly that releasing is a lot more complicated. If we like this mental model there are probably ways to make releasing easier (if we decide we don't like it, can totally go back)

**Pull request merged**